rag/flow/tests/test_title_chunker_token_cap.py

```
@@ -498,4 +498,48 @@ def test_tokenizer_failure_falls_back_to_char_count():
+    # path with the real tokenizer on multibyte, boundary-less text and asserts
+    # the concatenated sub-chunks reproduce the original text exactly (no
+    # character dropped at the split boundary).
+    import importlib.util
+
+    root = Path(__file__).resolve().parents[3]
+    spec = importlib.util.spec_from_file_location("ragflow_real_token_utils", str(root / "common" / "token_utils.py"))
+    real_token_utils = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(real_token_utils)
+
+    # Multibyte, no sentence boundary -> forces the hard-split fallback path.
+    body = "中文混合English文本再混入一些标点符号，" * 30
```

Ruff (0.16.1)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)